js/plugins/chroma/src/index.ts

```diff
@@ -120,7 +122,7 @@ export function chromaRetriever<
   createCollectionIfMissing?: boolean;
   embedder: EmbedderArgument<EmbedderCustomOptions>;
   embedderOptions?: z.infer<EmbedderCustomOptions>;
-}) {
+}): RetrieverAction<z.ZodOptional<typeof ChromaRetrieverOptionsSchema>> {
```

Comment:
This seemed a little funky to me. Not sure if this is 💯 .

It looks like we're possibly double-wrapping in an Optional? https://github.com/firebase/genkit/blob/main/js/ai/src/retriever.ts#L164-L168

Reply:
When I first made these abstractions in `js/ai`, I wanted to make as few assumptions as possible. That meant that the options should not be `ZodOptional` since we do not know what kind of retriever they are building and what options are required vs optional. 

IMO, the `/ai` abstractions should not add a `.optional()` on the schema and the providers can add them in the custom schema they define (`ChromaRetrieverOptionsSchema` in this case). 

That being said, this code has through many changes and its evident that it no longer holds to that thought. The problem here does not seem to be double wrapping though -- there is no Optional on the `ChromaRetrieverOptionsSchema` object and it is added by `RetrieverAction`.